packages/wrangler/src/cfetch/internal.ts

```diff
@@ -20,6 +20,23 @@ export async function fetchInternal<ResponseType>(
   init: RequestInit = {},
   queryParams?: URLSearchParams
 ): Promise<ResponseType> {
+  return responseToJson<ResponseType>(
+    await fetchInternalResponse(resource, init, queryParams)
+  );
+}
+
+/**
+ * An implementation detail of `fetchInternal` that does the actual fetching.
+ * We export this separately so that we can use it for one specific api,
+ * `kv:key get`/`getKeyValue()` in `cfetch/kv.ts`, because it's the only
+ * cloudflare api that doesn't return json in the standard FetchResult format.
+ */
+
+export async function fetchInternalResponse(
```

Comment:
I'm tempted to make this NOT generic and let it work ONLY for the kv get pattern, inlining the implementation.